### PR TITLE
Datecdesdatamigration

### PR DESCRIPTION
**Pull request by id2359**
This is no longer a data migration despite the branch name.
It is a script which should be run on the live site ( post a 2.0 install)
The script runs over Modjgo objects ( either cdes or history) and updates the data if it detects any "bad" date cde values. Only values of common data elements of type date are updated.
The script creates an instance of a TimesStripper class or HistoryTimeStripper class defined in utils.
There are unit tests on this class - initially I was thinking this would work in a data migration
so I added a backward step, but now I am only using the forward step in a transaction in the script.

### CODE REVIEW

**Comment by id2359** [comment]
ticket https://github.com/muccg/rdrf/issues/393

File: rdrf/rdrf/tests.py [hunk 1]
```diff
@@ -485,3 +486,569 @@ def test_date(self):
...
+            def __init__(self, pk, data):
+                self.pk = pk
+                self.data = data
+            def save(self):
+                print("Fake Modjgo save called")
+
+
+class TimeStripperTestCase(TestCase):
+    def setUp(self):
+        super(TimeStripperTestCase, self).setUp()
```
Comment (sztamas) [inline rdrf/rdrf/tests.py]: The super(CLASS, self).x(...) calls can all be changed to super().x(...) now that we're on python 3. 

Applies to all the similar calls.

File: rdrf/rdrf/utils.py [hunk 2]
```diff
@@ -441,3 +442,164 @@ def error(text):
...
+                            items = section["cdes"]
+                            for item in items:
+                                for cde in item:
+                                    if self.is_date_cde(cde):
+                                        if self.update_cde(cde):
+                                            updated += 1
+                                            
+        return updated > 0
+                    
+    def backward(self):
```
Comment (sztamas) [inline rdrf/rdrf/utils.py]: this can be deleted as discussed on Slack.

File: rdrf/rdrf/utils.py [hunk 3]
```diff
@@ -441,3 +442,164 @@ def error(text):
...
+    moving from Mongo to pure Django models with JSON fields ( "Modjgo" 0objects.)
+    CDE date values were converted  into iso strings including a time T substring.
+    This was done recursively for the cdes and history collections 
+    We provide forwards and backward methods to be used in a migration to fix this
+    I have used a class so we can unit test more easily.
+    """
+    
+    def __init__(self, dataset):
+        self.dataset = dataset # queryset live , lists of data records for testing
+        self.backup_data = {}
```
Comment (sztamas) [inline rdrf/rdrf/utils.py]: I think the contents of this (what has been updated) could be written out to a file after the script finished, so we have a record of what exactly has been updated.

A bit of extra precaution doesn't hurt.

File: rdrf/rdrf/utils.py [hunk 4]
```diff
@@ -441,3 +442,164 @@ def error(text):
 
     def __repr__(self):
         return self.text
+
+
+
+class TimeStripper(object):
+    """
+    This class exists to fix an error we introduced in the migration
+    moving from Mongo to pure Django models with JSON fields ( "Modjgo" 0objects.)
```
Comment (sztamas) [inline rdrf/rdrf/utils.py]: Also, can you please update the comment to not say this should be used from a migration.

**Pull request merged**